ocsmesh/hfun/raster.py

```diff
@@ -48,25 +49,26 @@ def __set__(self, obj, raster: Raster):
             raise TypeError(f'Argument raster must be of type {Raster}, not '
                             f'type {type(raster)}.')
         # init output raster file
-        # pylint: disable=R1732
-        tmpfile = tempfile.NamedTemporaryFile()
-        # TODO: Use contextlib.ExitStack
-        with rasterio.open(raster.tmpfile) as src:
+        with ExitStack() as stack:
+
+            src = stack.enter_context(rasterio.open(raster.tmpfile))
             if raster.chunk_size is not None:
                 windows = get_iter_windows(
                     src.width, src.height, chunk_size=raster.chunk_size)
             else:
                 windows = [rasterio.windows.Window(
                     0, 0, src.width, src.height)]
+
             meta = src.meta.copy()
             meta.update({'driver': 'GTiff', 'dtype': np.float32})
-            with rasterio.open(tmpfile, 'w', **meta,) as dst:
-                for window in windows:
-                    values = src.read(window=window).astype(np.float32)
-                    values[:] = np.finfo(np.float32).max
-                    dst.write(values, window=window)
+            dst = stack.enter_context(
+                    obj.modifying_raster(use_src_meta=False, **meta))
```

Comment:
HfunRaster object doesn't have any "path" attribute set, as a result if one tried to call `hfun_raster_obj.src` at this point it will fail. At this point in the code `src` and `tmpfile` are not set, any call to these attributes will result in a call to `path` which will fail in this case. Using helper argument `use_src_meta` we can avoid this issue when using the helper contextmanager method.